src/main/java/impl/weather/tempInfoComp.java

```diff
@@ -0,0 +1,12 @@
+package impl.weather;
+
+import api.weather.DayTemperatureInfo;
+
+import java.util.Comparator;
+
+public class tempInfoComp implements Comparator<DayTemperatureInfo> {
+    @Override
+    public int compare(DayTemperatureInfo o1, DayTemperatureInfo o2) {
+        return o1.getTemperature() - o2.getTemperature() - 1;
```

Comment:
Why do you subtract -1 here?

Reply:
Because of the way the comp works it will swap the places if they are equal and as long as we don't want that to happen we should subtract a small value eps to keep it working fine but since we are working with integers we can subtract 1.

Reply:
Did you mean how sorting works? Comparator just returns an integer no more. The default sorting algorithm in Java is merge sort, so it is a stable sorting algorithm

Reply:
You saw the wrong order due to another problem in your code - using HashMap